src/avtt/content-script.js

```diff
@@ -0,0 +1,580 @@
+        return match.split('\n').join('\n\n');
+    })
+}
+
+
+
+function advantageRoll(request, name, roll) {
+    return {
+        [RollType.NORMAL]: [[name, roll]],
+        [RollType.DOUBLE]: [[name, roll], ["2nd time", roll]],
+        [RollType.THRICE]: [[name, roll], ["2nd time", roll], ["3rd time", roll]],
+        [RollType.ADVANTAGE]: [[name, roll], ["Advantage", roll]],
+        [RollType.DISADVANTAGE]: [[name, roll], ["Disadvantage", roll]],
+        [RollType.SUPER_ADVANTAGE]: [[name, roll], ["Advantage", roll], ["Super Advantage", roll]],
+        [RollType.SUPER_DISADVANTAGE]: [[name, roll], ["Disadvantage", roll], ["Super Disadvantage", roll]],
+    }[request.advantage] || [[name, roll]]
+}
+
+function rollSkill(request, custom_roll_dice = "") {
+    let modifier = request.modifier;
+    // Custom skill;
+    if (request.ability === "--" && request.character.abilities.length > 0) {
+        let prof = "";
+        let prof_val = "";
+        let reliableTalent = false;
+        if (request.proficiency === "Proficiency") {
+            prof = "PROF";
+            prof_val += request.character.proficiency;
+            reliableTalent = request.reliableTalent;
+        } else if (request.proficiency === "Half Proficiency") {
+            prof = "HALF-PROFICIENCY";
+            prof_val += "floor(" + request.character.proficiency + " / 2)";
+        } else if (request.proficiency === "Expertise") {
+            prof = "EXPERTISE";
+            prof_val += `${request.character.proficiency} * 2`;
+            reliableTalent = request.reliableTalent;
+        }
+        const d20 = reliableTalent ? "1d20min10" : "1d20";
+        return {
+            icon: "bunker-soldier-standing",
+            color: "#7ED321",
+            title: request.skill + " Skill Check",
+            message: template([
+                ...advantageRoll(request, "Skill Check", `!(${d20}${prof_val.length > 0 ? " + " + prof_val : ""}${custom_roll_dice.length > 0 ? " + " + custom_roll_dice : ""})`)
+            ])
+        }
+        
+    } else {
+        let d20 = request.reliableTalent ? "1d20min10" : "1d20";
+        if (request.silverTongue && (request.skill === "Deception" || request.skill === "Persuasion"))
+            d20 = "1d20min10";
+
+        
+        return {
+            icon: "bunker-soldier-standing",
+            color: "#7ED321",
+            title: request.skill + " Skill Check",
+            message: template([
+                ...advantageRoll(request, "Skill Check", `!(${d20}${modifier}${custom_roll_dice.length > 0 ? " + " + custom_roll_dice : ""})`)
+            ])
+        }
+    }
+}
+
+function rollAbility(request, custom_roll_dice = "") {
+    return {
+        icon: "coin-star",
+        color: "#4A90E2",
+        title: request.ability + " Ability Check",
+        message:  template([
+            ...advantageRoll(request, "Ability Check", `!(1d20${request.modifier}${custom_roll_dice.length > 0 ? " + " + custom_roll_dice : ""})`)
```

Comment:
I see a lot of duplication here, I would suggest you make a function to generate the roll.
In some places you check that the modifier has length > 0, in others, you don't, you always do that code for the custom dice roll as well. You also add ` + ` in front of the custom roll dice, but you don't check if it's something that doesn't need it. For example a custom modifier of "- 2" or "+ 1"would cause it to become `1d20 + - 2` or `1d20 + + 1` which would both fail (I just tested it). So you'd need to trim the value, check if it starts with `+` or `- and add the `+` only if it doesn't.
Check either roll20 or the roll renderer code, I know I do that somewhere already.
As this becomes complicated, and a fix here would need to be copied in every other function, that's why having it separate to generate the roll would make sense.
Same comment from above about setting the variable elsewhere and not using the spread operator as it's not necessary, applies here and to every other similar line.